spinn_front_end_common/interface/interface_functions/preallocate_resources_for_extra_monitor_support.py

```diff
@@ -1,30 +1,43 @@
-from pacman.model.resources import CoreResource, \
+from pacman.model.resources import SpecificChipSDRAMResource, CoreResource, \
     PreAllocatedResourceContainer
+from pacman.model.resources.specific_board_iptag_resource import \
+    SpecificBoardTagResource
+from spinn_front_end_common.utility_models.\
+    data_speed_up_packet_gatherer_machine_vertex import \
+    DataSpeedUpPacketGatherMachineVertex
 from spinn_utilities.progress_bar import ProgressBar
+from spinnman.connections.udp_packet_connections import UDPConnection
 
 
 class PreAllocateResourcesForExtraMonitorSupport(object):
-    """ allocate resources needed for the extra monitor support
-    """
 
     def __call__(
             self, machine, pre_allocated_resources=None,
             n_cores_to_allocate=1):
-        """ setter offer
+        """
 
         :param machine: spinnaker machine object
         :param pre_allocated_resources: resources already pre allocated
         :param n_cores_to_allocate: config params for how many gatherers to use
         """
 
         progress = ProgressBar(
+            len(list(machine.ethernet_connected_chips)) +
             len(list(machine.chips)),
```

Comment:
This really ought to be `machine.n_chips`. (Also, we need to add a method to get the number of ethernet connected chips from the machine; we shouldn't get the list and copy it just to get the length.)

Reply:
fixed the n_chips, but until theres a n_ethernet chips, i dont think tis a blocker to this pr. given ethernt_connected_chips is a generator and so you need to lsit it to be able to get a length